src/storage/v3/splitter.hpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 Memgraph Ltd.
```

Comment:
Two things popped into my mind: the `Shard` has two members:
1. `deleted_vertices_`
2. `deleted_edges_`

I think they should be taken care of in some way. I see two ways:
1. make a note somewhere about they are not being handled. This is fine as currently we cannot delete edges or vertices.
2. Handle them.